src/server/chatController.py

```diff
@@ -0,0 +1,158 @@
+import cherrypy
+from cherrypy import request
+import simplejson
+from ws4py.websocket import WebSocket
+import uuid  # XXX Temp for fake session id... Could be used for real?
+
+
+class ChatController(object):
+  """
+  Controller of the chat
+  """
+
+  def __init__(self, app, logger):
+    """
+    ChatController initialiser
+
+    @type app: cide.app.python.chat.Chat
+    @type logger: logging.Logger
+
+    @param app: The Chat App instance
+    @param logger: The CIDE.py logger instance
+    """
+    self._app = app
+    self._logger = logger
+
+    self._logger.debug("ChatController instance created")
+
+  @cherrypy.expose
+  def connect(self):
+    """
+    Subscribe a client to the chat, to receive new messages
+    Method : PUT
+    (Path : /chat/connect)
+
+    The user may start to receive messages before he gets the response for this request.
+    """
+    if not cherrypy.session.get('username'):
+      cherrypy.session['username'] = str(uuid.uuid4())  # XXX Session should be set by the id module
+
+    username = cherrypy.session['username']
+    self._logger.info("Connect to chat requested by {0} ({1}:{2})".format(username,
+                                                                          request.remote.ip,
+                                                                          request.remote.port))
+
+    result = self._app.addUser(username)
+    self.sendTo(*result)
+
+  @cherrypy.expose
+  def disconnect(self):
+    """
+    Unsubscribe a client from the chat, stop sending new messages
+    Method : PUT
+    (Path : /chat/disconnect)
+    """
+    if not cherrypy.session.get('username'):
+      cherrypy.session['username'] = str(uuid.uuid4())  # XXX Session should be set by the id module
+
+    username = cherrypy.session['username']
+    self._logger.info("Disconnect from chat requested by {0} ({1}:{2})".format(username,
+                                                                               request.remote.ip,
+                                                                               request.remote.port))
+
+    result = self._app.removeUser(username)
+    self.sendTo(*result)
+
+  @cherrypy.expose
+  @cherrypy.tools.json_in()
+  def send(self):
+    """
+    Receive new message from the client
+    Method : PUT
+    (Path : /chat/send)
+
+    Input must be JSON of the following format:
+      {
+        'message': '<<Content of the message>>'
+      }
+
+    Output on the WS will be JSON of the following format:
+      {
+        'author':  '<<Author (username) of the sender>>',
+        'message': '<<Content of the message>>'
+      }
+    """
+    if not cherrypy.session.get('username'):
+      cherrypy.session['username'] = str(uuid.uuid4())  # XXX Session should be set by the id module
```

Comment:
This code appears many times.
It will be reviewed when integrating identification/authentication
:+1: